File: src/employeemanager/model/PayrollEmployee.java
package employeemanager.model;

import employeemanager.exceptions.InvalidDataException;

import java.time.LocalDate;

public class PayrollEmployee extends FullTimeEmployee {

    public PayrollEmployee(int employeeId,
                           String firstName,
                           String lastName,
                           LocalDate startDate,
                           LocalDate dateOfBirth,
                           double annualSalary,
                           int vacationDays,
                           String benefits) throws InvalidDataException {
        super(employeeId, firstName, lastName, startDate, dateOfBirth,
              annualSalary, vacationDays, benefits);
    }

    public void processPayroll(Employee employee) {
        System.out.println("Processing payroll for " + employee.getFullName() +
                " Amount: " + employee.getPay());
    }

    @Override
    public void reportToManager() {
        System.out.println("[Payroll] " + getFullName() + " reports to finance manager.");
    }
}

File: Part 3/EmployeeManager/models/PayrollEmployee.cs
using System;

namespace EmployeeManager.Models
{
    public class PayrollEmployee : FullTimeEmployee
    {
        public PayrollEmployee(int employeeId, string firstName, string lastName, DateTime startDate, DateTime dateOfBirth, double annualSalary, int vacationDays, string benefits) : base(employeeId, firstName, lastName, startDate, dateOfBirth, annualSalary, vacationDays, benefits) { }

        public void ProcessPayroll (Employee employee)
        {
            Console.WriteLine($"Processing payroll for {employee.FullName} Amount: {employee.GetPay()}");
        } 

        public override void ReportToManager() => Console.WriteLine($"[Payroll] {FullName} reports to the finance Manager");
    }
}